Firestore/Source/Local/FSTLevelDB.h

```diff
@@ -40,13 +41,16 @@ NS_ASSUME_NONNULL_BEGIN
 @interface FSTLevelDB : NSObject <FSTPersistence, FSTTransactional>
 
 /**
- * Initializes the LevelDB in the given directory. Note that all expensive startup work including
- * opening any database files is deferred until -[FSTPersistence start] is called.
+ * Creates a LevelDB in the given directory and sets `ptr` to point to it. Return value indicates
+ * success in creating the leveldb instance and must be checked before accessing `ptr`. C++ note:
+ * Once FSTLevelDB is ported to C++, this factory method should return StatusOr<>. It cannot
+ * currently do that because ObjC references are not allowed in StatusOr.
  */
-- (instancetype)initWithDirectory:(firebase::firestore::util::Path)directory
-                       serializer:(FSTLocalSerializer *)serializer
-                        lruParams:(firebase::firestore::local::LruParams)lruParams
-    NS_DESIGNATED_INITIALIZER;
++ (firebase::firestore::util::Status)dbWithDirectory:(firebase::firestore::util::Path)directory
```

Comment:
The Objective-C convention is to pass the error as an out parameter, but which convention applies here is confusing because it's a mix of C++ and Objective-C. Given that the return type itself is an Objective-C object pointer I'd propose that the Objective-C convention binds more tightly.

Feel free to ignore if you're disinclined to make the change--I recognize this class isn't going to live for too much longer :-). Note that API classes (`FIRFoo`) are going to live on so it's worth keeping an eye out for this kind of thing elsewhere even if you don't fix this here.

Reply:
I would prefer as a general rule we go with the C++ convention and annotate it with  `ABSL_MUST_USE_RESULT`. Otherwise, we could end up ignoring the error.

I'll add it here too.